src/components/SidePanel/PowerMode.jsx

```diff
@@ -50,6 +50,15 @@ export default () => {
     const { capabilities, isSmuMode, deviceRunning } = useSelector(appState);
     const togglePowerMode = () => dispatch(setPowerMode(!isSmuMode));
 
+    useEffect(() => {
+        const shouldBePoweredOn = !isSmuMode && !deviceRunning;
+        if (shouldBePoweredOn) {
+            setTimeout(() => {
+                dispatch(setDeviceRunning(true));
+            }, 10);
+        }
+    }, [isSmuMode, deviceRunning, dispatch]);
```

Comment:
I suggest to consider solving the automatic power on to be part of the action that toggles to ampere meter mode. I rather regard these kinds of logic to be part of the actions and not implementation detail that belongs to the components.

Reply:
Good feedback, I totally agree.